### PR TITLE
Enhancement: Jinja service support's now global namespace

### PR DESCRIPTION
**Pull request by mithunbharadwaj**

### CODE REVIEW

File: asabiris/formatter/jinja/service.py [hunk 1]
```diff
@@ -28,6 +29,8 @@ def __init__(self, app, service_name="JinjaService"):
 			self.Variables = {}
 
 		self.Environment = jinja2.Environment()
+		# Inject 'now' function into the Jinja2 template's global namespace
+		self.Environment.globals['now'] = lambda: datetime.datetime.utcnow()
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:33]: `now` contains the start time of the service?

That's not correct.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py (hunk 1 above)
Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py:33]: @ateska From the timestamp they have different time generated at different times

Screenshot 2024-06-07 at 2 19 56 PM

File: asabiris/formatter/jinja/service.py [hunk 2]
```diff
@@ -28,8 +30,15 @@ def __init__(self, app, service_name="JinjaService"):
 			self.Variables = {}
 
 		self.Environment = jinja2.Environment()
+		# Inject 'now' function into the Jinja2 template's global namespace
+		self.Environment.globals['now'] = self._get_current_time
 		self._load_variables_from_json()
 
+	def _get_current_time(self):
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:37]: 1) Name `_jinja2_now` ...
2) It should return `current_time` ... a formatting should be done in Jinja2 template

**Review by ateska — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py [hunk 3]
```diff
@@ -28,8 +30,15 @@ def __init__(self, app, service_name="JinjaService"):
 			self.Variables = {}
 
 		self.Environment = jinja2.Environment()
+		# Inject 'now' function into the Jinja2 template's global namespace
+		self.Environment.globals['now'] = self._jinja_now
 		self._load_variables_from_json()
 
+	def _jinja_now(self):
+		timezone = pytz.timezone('Europe/Prague')
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:38]: Timezone must be taken from the tenant configuration, cannot be hardcoded.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py (hunk 3 above)
Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py:38]: @ateska Okay, how do I extract timezone from tenant configuration?

Comment (ateska) [reply asabiris/formatter/jinja/service.py:38]: I guess that you know the tenant (if you don't then fallback to UTC timezone).
Then you need to query SeaCat Auth API ... using discover service.

**Pull request merged**